From 4ed35e7d81b2cb0758c055cdfc9ac206636504ea Mon Sep 17 00:00:00 2001
From: Takahiro Ueda <t.ueda.od@juntendo.ac.jp>
Date: Wed, 25 Mar 2026 22:36:10 +0900
Subject: [PATCH] fix(emela): avoid crash when Boost directory is not detected

Avoid indexing an empty list when no Boost directory is detected under
the HEPTools prefix directory. eMELA's CMake configuration may still
be able to find Boost, e.g., from a system-wide installation.
---
 HEPToolInstaller.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/HEPToolInstaller.py b/HEPToolInstaller.py
index 78c4f05..b8ece4a 100755
--- a/HEPToolInstaller.py
+++ b/HEPToolInstaller.py
@@ -944,7 +944,8 @@ def install_emela(tmp_path):
     """Installation operat ons for COLLIER"""
     log = open(pjoin(_HepTools['emela']['install_path'],"emela_install.log"), "w")
     my_env = os.environ.copy()
-    boost_dir = [p for p in os.listdir(_HepTools['boost']['install_path']) if p.startswith('boost') and not p.endswith('.log')][0]
+    boost_dirs = [p for p in os.listdir(_HepTools['boost']['install_path']) if p.startswith('boost') and not p.endswith('.log')]
+    boost_dir = boost_dirs[0] if boost_dirs else '/usr/include'
 
 
     p = subprocess.call([pjoin(_installers_path,'installEMELA.sh'),
-- 
2.34.1

